types/joi/index.d.ts

```diff
@@ -825,20 +844,26 @@ export interface Terms {
 export interface Rules {
     name: string;
     params?: ObjectSchema | { [key: string]: Schema };
-    setup?: Function;
-    validate?: Function;
+    setup?: (this: AnySchema<AnySchema<Schema>>, description: { [key: string]: any }) => AnySchema<AnySchema<Schema>> | void;
+    validate?: (this: AnySchema<AnySchema<Schema>>, params: { [key: string]: any }, value: any, state: State, options: ValidationOptions) => Err | undefined;
```

Comment:
Basically, I'm still somewhat confused by the self referencing interface:
```export interface AnySchema<T extends AnySchema<Schema>>``` but this appears fulfill the interface, I could have probably used `any`, but I hoped for some clarification from the original authors.